src/diagram.js

```diff
@@ -89,3 +89,189 @@ export function toTeX(diagram) {
         </Diagram>
     )
 }
+
+export function fromTeX(code) {
+    let consumers = [
```

Comment:
Yes :D There was a reason for this: for me, `consumers` was "special" because it contains the actual logic and thus I wanted to have it at the top. However, I have never thought about what would happen if one tried to read the code starting from the top. Should be fixed in a second.